src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -244,7 +244,7 @@ public function dump(array $options = []): string|array
         if ($this->addGetService) {
             $code = preg_replace(
                 "/(\r?\n\r?\n    public function __construct.+?\\{\r?\n) ++([^\r\n]++)/s",
-                "\n    protected \Closure \$getService;$1        \$containerRef = $2\n        \$this->getService = static function () use (\$containerRef) { return \$containerRef->get()->getService(...\\func_get_args()); };",
+                "\n    protected \Closure \$getService;$1        \$containerRef = $2\n        \$this->getService = function () use (\$containerRef) { return \$containerRef->get()->getService(...\\func_get_args()); };",
```

Comment:
this one does not look like a closure being in a static method, as it has `$this->getService = `

Reply:
If it's needed, here it is: https://github.com/symfony/symfony/pull/49476